File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
internal class Lasagna
{
	private const int _expectedCookingTime = 40;
	private const int _preparationMinutesPerLayer = 2;

	internal int ElapsedTimeInMinutes(int layerCount, int elapsedTime) =>
		elapsedTime + PreparationTimeInMinutes(layerCount);

	internal int ExpectedMinutesInOven() =>
		_expectedCookingTime;

	internal int PreparationTimeInMinutes(int layerCount) =>
		layerCount * _preparationMinutesPerLayer;

	internal int RemainingMinutesInOven(int elapsedTime) =>
		_expectedCookingTime - elapsedTime;
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna
let expectedMinutesInOven = 40
let remainingMinutesInOven(minutesAlreadyInOven: int): int =  expectedMinutesInOven - minutesAlreadyInOven

let minutesToPrepareEachLayer = 2
let preparationTimeInMinutes(layerCount: int): int = minutesToPrepareEachLayer * layerCount

let elapsedTimeInMinutes(layerCount: int)(minutesAlreadyInOven: int): int = preparationTimeInMinutes(layerCount) + minutesAlreadyInOven
